### PR TITLE
#49: Adding unscheduled maintenance endpoint.

### PR DESCRIPTION
**Pull request by MS-ResilienX**

### CODE REVIEW

File: MaintenanceApi.yaml [hunk 1]
```diff
@@ -17,6 +17,8 @@ tags:
   
 - name: Maintenance Component
   description: Maintenance Component data model endpoints  
+- name: Maintenance (Unscheduled)
```
Comment (GD-ResilienX) [inline MaintenanceApi.yaml:20]: This doesn't line up with the tags called out in the unscheduled maintenance endpoints

File: MaintenanceApi.yaml [hunk 2]
```diff
@@ -425,7 +427,123 @@ paths:
       security:
       - MaintenanceDefinitionAuth:
         - fraihmwork.maintenance.write
-                
+  /maintenance/unscheduled:
+    get:
+      tags:
+        - Unscheduled Maintenance
```
Comment (GD-ResilienX) [inline MaintenanceApi.yaml:433]: Should line up with the tags set up top (see other issue)

**Review by GD-ResilienX — CHANGES_REQUESTED** [review]
Swagger tidiness issues :(

**Review by GD-ResilienX — APPROVED** [review]
👍

**Comment by AS-resilienx** [comment]
Unscheduled Maintenance -> GET -> should we add a 404 response for an invalid component id?

**Comment by MS-ResilienX** [comment]
> Unscheduled Maintenance -> GET -> should we add a 404 response for an invalid component id?

What's the precedent? Do we individually validate query parameters?

File: MaintenanceApi.yaml [hunk 3]
```diff
@@ -425,7 +427,123 @@ paths:
...
+          description: Uuid of the component's unscheduled maintenance being requested
+        - in: query
+          name: state
+          schema:
+            type: string
+            format: UnscheduledMaintenanceState
+          required: false
+          description: UnscheduledMaintenanceState of the unscheduled maintenance being requested
+      responses:
+        200:
```
Comment (GD-ResilienX) [inline MaintenanceApi.yaml:453]: Needs a 400 bad request for bad query parameters

File: MaintenanceApi.yaml [hunk 4]
```diff
@@ -425,7 +427,129 @@ paths:
       security:
       - MaintenanceDefinitionAuth:
         - fraihmwork.maintenance.write
-                
+  /maintenance/unscheduled:
```
Comment (GD-ResilienX) [inline MaintenanceApi.yaml:430]: Is there a reason that *everything* here is under a maintenance path? The spec already says everything is under api/maintenance/v0?

Comment (GD-ResilienX) [reply MaintenanceApi.yaml:430]: This will be handled in future issue

**Review by GD-ResilienX — APPROVED** [review]
Back to approval

**Pull request merged**